selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceProjectsSamples.java

```diff
@@ -0,0 +1,178 @@
+    ADD_BUTTON("add-project-button");
+
+    private final String buttonId;
+
+    BottomButton(String buttonId) {
+      this.buttonId = buttonId;
+    }
+
+    public String get() {
+      return this.buttonId;
+    }
+  }
+
+  public void waitSamplesForm() {
+    waitButtons(
+        asList(
+            SAMPLES_BUTTON,
+            BLANK_BUTTON,
+            GIT_BUTTON,
+            GITHUB_BUTTON,
+            ZIP_BUTTON,
+            CANCEL_BUTTON,
+            ADD_BUTTON));
+  }
+
+  public WebElement waitButton(Button button) {
+    return seleniumWebDriverHelper.waitVisibility(By.id(button.get()));
+  }
+
+  public void clickOnButton(Button button) {
+    waitButton(button).click();
+  }
+
+  public void waitTabSelected(UpperButton upperButton) {
+    String tabXpath = format(TAB_STATE_TEMPLATE_XPATH, upperButton.get());
+    seleniumWebDriverHelper.waitAttributeContainsValue(
+        By.xpath(tabXpath), "class", "che-toggle-button-enabled");
+  }
+
+  public void waitTabNotSelected(UpperButton upperButton) {
+    String tabXpath = format(TAB_STATE_TEMPLATE_XPATH, upperButton.get());
+    seleniumWebDriverHelper.waitAttributeContainsValue(
+        By.xpath(tabXpath), "class", "che-toggle-button-disabled");
+  }
+
+  public void waitButtons(List<Button> buttons) {
+    buttons.forEach(button -> waitButton(button));
+  }
+
+  public void waitSampleByName(String... sampleNames) {
+    asList(sampleNames)
+        .forEach(
+            name -> {
+              String sampleXpath = format(SAMPLE_ITEM_TEMPLATE_XPATH, sampleNames);
+              seleniumWebDriverHelper.waitVisibility(By.xpath(sampleXpath));
+            });
+  }
+
+  public boolean isCheckboxEnabled(String sampleName) {
+    String checkboxId = format(CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE, sampleName);
+    String checkboxLocator = format("//div[@id='%s']/md-checkbox", checkboxId);
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(By.xpath(checkboxLocator), "aria-checked")
+        .equals("true");
+  }
+
+  public void waitCheckboxEnabled(String... sampleNames) {
+    asList(sampleNames)
+        .forEach(
+            name ->
+                seleniumWebDriverHelper.waitSuccessCondition(driver -> isCheckboxEnabled(name)));
+  }
+
+  public void waitCheckboxDisabled(String... sampleNames) {
+    asList(sampleNames)
+        .forEach(
+            name ->
+                seleniumWebDriverHelper.waitSuccessCondition(driver -> !isCheckboxEnabled(name)));
+  }
+
+  public void clickOnCheckbox(String... sampleNames) {
+    asList(sampleNames)
+        .forEach(
+            name -> {
+              String checkboxId = format(CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE, name);
+              String checkboxLocator = format("//div[@id='%s']/md-checkbox", checkboxId);
+              seleniumWebDriverHelper.waitAndClick(By.xpath(checkboxLocator));
+            });
+  }
+
+  private void waitButtonState(BottomButton bottomButton, boolean state) {
```

Comment:
**BottomButton** looks strange here so as it has been described in another class whereas this class has `UpperButton`.